components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.ts

```diff
@@ -0,0 +1,136 @@
+    this.loading$ = store.pipe(select(getStatus), map(loading));
+
+    this.sortedChefServers$ = store.pipe(
+      select(allServers),
+      map(servers => ChefSorters.naturalSort(servers, 'name')));
+
+    this.createChefServerForm = this.fb.group({
+      // Must stay in sync with error checks in create-chef-server-modal.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      description: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      fqdn: ['', [Validators.required,
+        Validators.pattern(Regex.patterns.NON_BLANK),
+        Validators.pattern(Regex.patterns.VALID_FQDN)
+      ]],
+      ip_address: ['', [Validators.required,
+        Validators.pattern(Regex.patterns.NON_BLANK),
+        Validators.pattern(Regex.patterns.VALID_IP_ADDRESS)
+      ]]
+    });
+  }
+
+  ngOnInit() {
+    this.layoutFacade.showInfrastructureSidebar();
+    this.store.dispatch(new GetServers());
+    this.store.pipe(
+      select(saveStatus),
+      takeUntil(this.isDestroyed),
+      filter(state => this.createModalVisible && !pending(state)))
+      .subscribe(state => {
+        this.creatingChefServer = false;
+        if (state === EntityStatus.loadingSuccess) {
+          this.closeCreateModal();
+        }
+      });
+
+    combineLatest([
+      this.store.select(saveStatus),
+      this.store.select(saveError)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      filter(() => this.createModalVisible),
+      filter(([state, error]) => state === EntityStatus.loadingFailure && !isNil(error)))
+      .subscribe(([_, error]) => {
+        if (error.status === HttpStatus.CONFLICT) {
+          this.conflictErrorEvent.emit(true);
+        } else {
+          // Close the modal on any error other than conflict and display in banner.
+          this.closeCreateModal();
+        }
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  public openCreateModal(): void {
+    this.createModalVisible = true;
+    this.resetCreateModal();
+  }
+
+  public closeCreateModal(): void {
+    this.createModalVisible = false;
+    this.resetCreateModal();
+  }
+
+  public createChefServer(): void {
+    this.creatingChefServer = true;
+    const server = {
+      name: this.createChefServerForm.controls['name'].value.trim(),
+      description: this.createChefServerForm.controls['description'].value.trim(),
+      fqdn: this.createChefServerForm.controls['fqdn'].value.trim(),
+      ip_address: this.createChefServerForm.controls['ip_address'].value.trim()
+    };
+    this.store.dispatch(new CreateServer(server));
+  }
+
+  private resetCreateModal(): void {
+    this.creatingChefServer = false;
+    this.createChefServerForm.reset();
+    this.conflictErrorEvent.emit(false);
+  }
+
+  public startServerDelete(server: Server): void {
+    this.serverToDelete = server;
+    this.deleteModalVisible = true;
+  }
```

Comment:
related to my comment above, since we are now using `mat-select` instead of `chef-control-menu`, we need to put a guard in place or we'll get two event fires.  Suggest changing this function to look like this:

```
  public startServerDelete($event: ChefKeyboardEvent, server: Server): void {
    if ($event.isUserInput) {
      this.serverToDelete = server;
      this.deleteModalVisible = true;
    }
  }
```

The reason for this is that when using mat-select, if fires off two events, one for a selection and one for a deselection.  `$event.isUserInput = true` is the user selected event.

Reply:
@SEAjamieD Thanks for the review. I have incorporated all suggested changes please have a look

Reply:
Thanks for the changes - they look great 👍